### PR TITLE
Test:UnitTest for  BLOCK_TASK_DIST_1D

### LINKED ISSUE #2195: I want to  add the UnitTest for  BLOCK_TASK_DIST_1D
### Describe Current Status and Possible Solution

UnitTest for  BLOCK_TASK_DIST_1D is lacking.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by SeptemberMy**
fix #2195

### CODE REVIEW

File: source/module_base/test/tool_threading_test.cpp [hunk 1]
```diff
@@ -8,6 +8,8 @@
 * - Tested functions of class threading tool:
 *   - TASK_DIST_1D:
 *       - (template)Distributing 1d tasks by worker id (int and long long)
+*   - BLOCK_TASK_DIST_1D:
+*       - (template)Distributing 1d tasks by block_size and worker id (int and long long)
 **/
 class tool_threading_test : public testing::Test
```
Comment (hongriTianqi) [inline source/module_base/test/tool_threading_test.cpp]: This class seems useless here.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_base/test/tool_threading_test.cpp [hunk 2]
```diff
@@ -54,3 +56,47 @@ TEST(tool_threading, TASK_DIST_1D_test_longlong)
     EXPECT_EQ(st,6);
     EXPECT_EQ(le,6);
 }
+
+TEST(tool_threading, BLOCK_TASK_DIST_1D_test_int)
```
Comment (hongriTianqi) [inline source/module_base/test/tool_threading_test.cpp]: Underscores are not suggested to be used in naming unit tests. Hump-shaped naming style is preferred.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_base/test/tool_threading_test.cpp [hunk 3]
```diff
@@ -54,3 +56,47 @@ TEST(tool_threading, TASK_DIST_1D_test_longlong)
...
+
+TEST(tool_threading, BLOCK_TASK_DIST_1D_test_longlong)
+{
+    long long nw=1;
+    long long iw=1;
+    long long nt=16;
+    long long st;
+    long long le;
+    long long bs=1;
+    ModuleBase:: BLOCK_TASK_DIST_1D(nw,iw,nt,bs,st,le);
```
Comment (hongriTianqi) [inline source/module_base/test/tool_threading_test.cpp:95]: Could you please give short comments on the meaning of the parameters?

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_base/test/tool_threading_test.cpp (hunk 1 above)
Comment (SeptemberMy) [reply source/module_base/test/tool_threading_test.cpp]: > This class seems useless here.

Indeed, I didn't distinguish between TEST and TEST_F.

File: source/module_base/test/tool_threading_test.cpp (hunk 2 above)
Comment (SeptemberMy) [reply source/module_base/test/tool_threading_test.cpp]: > Underscores are not suggested to be used in naming unit tests. Hump-shaped naming style is preferred.

OK. I'll change it.

File: source/module_base/test/tool_threading_test.cpp (hunk 3 above)
Comment (SeptemberMy) [reply source/module_base/test/tool_threading_test.cpp:95]: > Could you please give short comments on the meaning of the parameters?

They are abbreviations for function parameter names.
nw: nworker
iw: iworker
nt: ntask
st: start
le: len
bs: block_size

Comment (hongriTianqi) [reply source/module_base/test/tool_threading_test.cpp:95]: Adding them into the test file as comments will facilitate future review.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_base/test/tool_threading_test.cpp (hunk 3 above)
Comment (SeptemberMy) [reply source/module_base/test/tool_threading_test.cpp:95]: > Adding them into the test file as comments will facilitate future review.
OK.

**Review by hongriTianqi — APPROVED** [review]

**Pull request merged**